# E-commerce Sales Analysis

**Python | pandas | Exploratory Data Analysis**

## 1. Project Overview

The goal of this project is to analyze e-commerce sales data to understand revenue performance, product demand, and category performance.

The analysis focuses on identifying sales trends, calculating key business metrics, and extracting actionable insights from the dataset.

## 2. Dataset Description

The dataset contains e-commerce order information with the following columns:

* `Order_ID`: Unique identifier for each order.
* `Date`: Date of the order.
* `Product`: Name of the purchased product.
* `Category`: Product category.
* `Quantity`: Number of units purchased.
* `Unit_Price`: Price per unit.
* `Revenue`: Total revenue generated by each order.

**Note:** This is a synthetic dataset created for educational purposes.

## 3. Data Exploration

The dataset was explored to understand its structure, dimensions, data types, and basic statistical properties.

## 4. Sales Analysis

Key business metrics were calculated, including:

* Total revenue.
* Total units sold.
* Average order value.
* Daily revenue.

## 5. Product and Category Analysis

Sales performance was analyzed across individual products and product categories to identify the highest-performing segments.

## 6. Key Findings

The analysis revealed differences between product sales volume and revenue contribution, highlighting the importance of evaluating business performance using multiple metrics.


In [1]:
import pandas as pd

In [2]:
data = {
    "Order_ID": [1001, 1002, 1003, 1004, 1005, 1006, 1007, 1008, 1009, 1010],
    "Date": [
        "2026-01-03", "2026-01-03", "2026-01-04", "2026-01-05",
        "2026-01-05", "2026-01-06", "2026-01-07", "2026-01-07",
        "2026-01-08", "2026-01-09"
    ],
    "Product": [
        "Laptop", "Mouse", "Keyboard", "Laptop", "Monitor",
        "Mouse", "Keyboard", "Laptop", "Monitor", "Mouse"
    ],
    "Category": [
        "Electronics", "Accessories", "Accessories", "Electronics",
        "Electronics", "Accessories", "Accessories", "Electronics",
        "Electronics", "Accessories"
    ],
    "Quantity": [1, 2, 1, 1, 2, 3, 2, 1, 1, 2],
    "Unit_Price": [900, 25, 45, 900, 250, 25, 45, 900, 250, 25]
}

df = pd.DataFrame(data)

df["Revenue"] = df["Quantity"] * df["Unit_Price"]

df.to_csv("sales.csv", index=False)

print("Το dataset δημιουργήθηκε επιτυχώς!")

Το dataset δημιουργήθηκε επιτυχώς!


In [3]:
df

,Order_ID,Date,Product,Category,Quantity,Unit_Price,Revenue
0,1001,2026-01-03,Laptop,Electronics,1,900,900
1,1002,2026-01-03,Mouse,Accessories,2,25,50
2,1003,2026-01-04,Keyboard,Accessories,1,45,45
3,1004,2026-01-05,Laptop,Electronics,1,900,900
4,1005,2026-01-05,Monitor,Electronics,2,250,500
5,1006,2026-01-06,Mouse,Accessories,3,25,75
6,1007,2026-01-07,Keyboard,Accessories,2,45,90
7,1008,2026-01-07,Laptop,Electronics,1,900,900
8,1009,2026-01-08,Monitor,Electronics,1,250,250
9,1010,2026-01-09,Mouse,Accessories,2,25,50


In [4]:
total_revenue = df["Revenue"].sum()
print("Total revenue:",total_revenue,"€")

Total revenue: 3760 €


In [9]:
total_quantity=df["Quantity"].sum()
print("Total products sold:",total_quantity)

Total products sold: 16


In [10]:
average_value=df["Revenue"].mean()
print("Average order value:",average_value, "€")

Average order value: 376.0 €


In [12]:
sales_by_product=df.groupby("Product")["Quantity"].sum()
print(sales_by_product)

Product
Keyboard    3
Laptop      3
Monitor     3
Mouse       7
Name: Quantity, dtype: int64


In [14]:
revenue_by_category=df.groupby("Category")["Revenue"].sum()
print(revenue_by_category)

Category
Accessories     310
Electronics    3450
Name: Revenue, dtype: int64


## Key Findings

Based on the analysis of the e-commerce sales dataset, the following findings were identified:

1. The dataset contains 10 orders and 16 units sold.
2. Total revenue amounts to €3,760.
3. The average order value is €376.
4. Mouse is the best-selling product by quantity, with 7 units sold.
5. Laptop generates the highest revenue, contributing €2,700.
6. Electronics is the leading category, generating €3,450, or approximately 91.8% of total revenue.
7. January 5, 2026, recorded the highest daily revenue, with €1,400.

### Conclusion

The analysis shows that product popularity by quantity does not necessarily correspond to revenue generation. Although Mouse has the highest sales volume, Laptop contributes the most revenue due to its higher unit price.

These findings demonstrate the importance of combining sales volume and revenue metrics when evaluating business performance.
